src/logic/game.cc

```diff
@@ -216,6 +218,46 @@ void Game::save_syncstream(bool const save) {
 	syncwrapper_.syncstreamsave_ = save;
 }
 
+void Game::check_addons_desync_magic() {
+	if (did_check_addons_desync_magic_) {
+		postload_addons();
+		return;
+	}
+	did_check_addons_desync_magic_ = true;
+
+	// TODO(Nordfriese): Minimal-invasive fix for a very evil desync. The problem is:
+	// - We load only the tribes actually being played when starting a game.
+	// - With some add-ons, it is necessary to load *all* tribes during loading the replay.
+	// - This results in different load orders, so the same item may have a different
+	//   DescriptionIndex in the replay than in the original game. In particular, it
+	//   may happen that the index of A is smaller than B's in the original game but
+	//   greater than B's in the replay.
+	// The simple (and ugly) solution is to load all tribes in the original game if
+	// we know that we'll need to do this in the replay as well. We thus get the
+	// same load order – at the cost of a much longer loading time, so this is not
+	// suited as a long-term solution. In the long run, it would probably be best
+	// to get rid of `DescriptionIndex` entirely in favour of `std::string`.
+
+	bool needed = false;
+	for (const AddOns::AddOnInfo& a : enabled_addons()) {
+		if (a.category == AddOns::AddOnCategory::kWorld ||
+		    a.category == AddOns::AddOnCategory::kTribes) {
+			needed = true;
+			break;
+		}
+	}
+	if (!needed) {
+		return;
+	}
+
+	delete_world_and_tribes();
+	descriptions();
+	// Cyclic dependency. Can and must be gotten rid of when fixing the above TO-DO.
+	EditorInteractive::load_world_units(nullptr, *this);
```

Comment:
I did a quick test without this line and encountered no problems. What is the intention here?